sda-commons-starter-s3/src/test/java/org/sdase/commons/spring/boot/s3/config/S3HealthIndicatorIntegrationTest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2022- SDA SE Open Industry Solutions (https://www.sda.se)
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+package org.sdase.commons.spring.boot.s3.config;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+
+import org.junit.jupiter.api.Test;
+import org.junitpioneer.jupiter.SetSystemProperty;
+import org.sdase.commons.spring.boot.s3.S3TestApp;
+import org.sdase.commons.spring.boot.web.testing.s3.S3Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.test.web.server.LocalManagementPort;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.annotation.DirtiesContext;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.HeadBucketRequest;
+
+@SetSystemProperty(key = "management.health.s3.enabled", value = "true")
+@SpringBootTest(
+    classes = S3TestApp.class,
+    webEnvironment = WebEnvironment.RANDOM_PORT,
+    properties = {"auth.disable=true", "opa.disable=true", "management.server.port=8071"})
+@S3Test
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
```

Comment:
NIT: Can the dirty context be avoided if we don't use SetSystemProperty? Afaik, this slows down the test suite.

Reply:
The problem is that I want to test when it is enabled and when it is not, so I need to run the server again for each test. In order to do that I either use this approach or I separate in two test classes, one with the health check enabled, and the other one disabled.